src/Pyz/Yves/CheckoutPage/Theme/default/views/payment/payment.twig

```diff
@@ -79,8 +79,7 @@
                     } only %}
                         {% block fieldset %}
                             <div class="js-payment-method-{{ embed.index }} {% if embed.index != '1-0' %} is-hidden{% endif %}">
-                                <h2 class="title title--primary">{{ embed.toggler.vars.label | trans }}</h2>
-
+                                <h2 class="title title--primary {% if (embed.index != '1-0') and (embed.index != '2-0') %} is-hidden{% endif %}">{{ embed.toggler.vars.label | trans }}</h2>
```

Comment:
so when we got new payment method we had to add `and (embed.index != '3-0')` ?
and then `and (embed.index != '4-0')`?

maybe we can solve it somehow differently but once and forever?

Reply:
@matweew 

I checked all the payment methods available but didn't find anything unique to decide whether to hide or show the labels for that particular payment method. Currently to hide the labels we have to add AND for each new payment method.

For example we have payment method name which combines provider(Payone), payment method & language like:
- payone-credit-card-de

we may have more payment method names as above.

![Payment method details](https://user-images.githubusercontent.com/100114749/168569807-40436745-e709-4c68-b156-7d8a9cb3b775.png)

Could you please suggest an identifier for different Payone payment methods e.g.
- Paypal
- Credit Card

Reply:
Do we have access to PaymentTransfer of each payment form?

it should have name something like `foreignPayments[payment-key]`


also PaymentTransfer->getForeignPayments() maybe it can be used to identify hidden payment methods except with key `1-0`

cc @pushokwhite

Reply:
I didn't check this code, but kinda `{% if embed.toggler.vars.value matches '#foreignPayments\[[a-zA-Z0-9_-]+\]#' %}` should be in this case.

Reply:
@matweew @pushokwhite 

I found **foreignPayments[payone-credit-card-de]** in **embed.toggler.vars.value** but there could be following cases:

1. payone-credit-card-de
2. payone-credit-card-at
3. payone-paypal-de
4. payone-paypal-at

Reply:
@pushokwhite I used regex as you suggested and its working fine.

I put condition on parent div instead of h2 tag.